axelor-base/src/main/java/com/axelor/apps/base/service/AddressServiceImpl.java

```diff
@@ -341,4 +351,60 @@ public String getZipCode(Address address) {
     }
     return null;
   }
+
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public void setFormattedFullName(Address address) {
+        computeStringAddressTemplate(address);
+  }
+
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public int computeFormattedAddressForCountries(List<Long> countryIds) {
+        int totalAddressFormatted = 0;
+        AddressRepository addressRepository = Beans.get(AddressRepository.class);
+        int offset = 0;
+        Query<Address> query =
+                    addressRepository.all().filter("self.addressL7Country.id in ?1", countryIds);
+
+                List<Address> addressList = query.fetch(FETCH_LIMIT, offset);
+
+        while (!addressList.isEmpty()) {
+            totalAddressFormatted = totalAddressFormatted + addressList.size();
+            generateFormattedAddressForAddress(addressList);
+
+                    JPA.clear();
+
+                    offset = addressList.size();
+            addressList = query.fetch(FETCH_LIMIT, offset);
+          }
+
+                return totalAddressFormatted;
```

Comment:
you should use a try catch and generate TraceBacks for each exception in this while loop
Also count the number of exceptions and return it with the number of formatted addresses. The user needs to know if errors happened.